instrumentation/opentelemetry-instrumentation-celery/pyproject.toml

```diff
@@ -37,7 +37,6 @@ instruments = [
 test = [
   "opentelemetry-instrumentation-celery[instruments]",
   "opentelemetry-test-utils == 0.38b0.dev",
-  "importlib-metadata==4.13.0",
```

Comment:
Is this `importlib-metadata` not used at all in this package?

Reply:
No, IIRC, this was pinned to fix the CI that was failing when `importlib-metadata` 5.x was released with breaking changes. This is also part of the `test` dependencies.

Reply:
The commit that added this https://github.com/open-telemetry/opentelemetry-python-contrib/commit/93b8398b6068728f999247ad53275d487374f116 to fix https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1371

Reply:
But the `opentelemetry-api` pinned version, however, feels very strict `importlib-metadata~= 6.0.0` and can potentially conflict with user programs?

Reply:
@srikanthccv 
it's possible. Maybe we can fix this on an ongoing basis if it comes up?

Reply:
> this is also part of the test dependencies.

ack that makes sense

Reply:
> But the opentelemetry-api pinned version, however, feels very strict importlib-metadata~= 6.0.0 and can potentially conflict with user programs?

I think there was a miscommunication with my comment on @ocelotl's PR. I agree we should do as wide as possible, I just wanted there to be *some* upper bound (which we can then expand as needed)

Reply:
I don't think this should be a blocker.